HLTrigger/Configuration/python/customizeHLTforAlpaka.py

```diff
@@ -1,5 +1,255 @@
 import FWCore.ParameterSet.Config as cms
+from HeterogeneousCore.AlpakaCore.functions import *
 
+## PF HLT in Alpaka
+def customizeHLTforAlpakaParticleFlowClustering(process):
+    '''Customization to introduce Particle Flow Reconstruction in Alpaka
+    '''
+    process.hltPFRecHitHCALParamsRecordSource = cms.ESSource('EmptyESSource',
+            recordName = cms.string('PFRecHitHCALParamsRecord'),
+            iovIsRunNotTime = cms.bool(True),
+            firstValid = cms.vuint32(1)
+            )
+
+    process.hltPFRecHitHCALTopologyRecordSource = cms.ESSource('EmptyESSource',
+            recordName = cms.string('PFRecHitHCALTopologyRecord'),
+            iovIsRunNotTime = cms.bool(True),
+            firstValid = cms.vuint32(1)
+            )
+
+    process.hltPFClusterParamsRecordSource = cms.ESSource('EmptyESSource',
+            recordName = cms.string('JobConfigurationGPURecord'),
+            iovIsRunNotTime = cms.bool(True),
+            firstValid = cms.vuint32(1)
+            )
+
+    process.hltHBHERecHitToSoA = cms.EDProducer("HCALRecHitSoAProducer@alpaka",
+            src = cms.InputTag("hltHbhereco"),
+            synchronise = cms.untracked.bool(False)
+            )
+
+    process.hltPFRecHitHCALTopologyESProducer = cms.ESProducer('PFRecHitHCALTopologyESProducer@alpaka',
+      usePFThresholdsFromDB = cms.bool(True),
+      appendToDataLabel = cms.string(''),
+    )
+
+
+    process.hltPFRecHitHCALParamsESProducer = cms.ESProducer('PFRecHitHCALParamsESProducer@alpaka',
+      energyThresholdsHB = cms.vdouble(
+        0.1,
+        0.2,
+        0.3,
+        0.3
+      ),
+      energyThresholdsHE = cms.vdouble(
+        0.1,
+        0.2,
+        0.2,
+        0.2,
+        0.2,
+        0.2,
+        0.2
+      ),
+      appendToDataLabel = cms.string(''),
+    )
```

Comment:
I thought these numbers should now be taken from the GT, so is this ESProducer still needed ? (I'm not thinking about the customisation per se, but about what we will actually integrate in the HLT menu)

Reply:
My understanding is that, yes, these numbers are now taken from GT, but we still need to provide some "default" numbers to get the code function. Perhaps something we can try to see if we can design the code differently.

Reply:
When adding DB thresholds, my understanding was to maintain function with hardcoded parameters if needed. An example of the current implementation starts here: https://github.com/cms-sw/cmssw/blob/82470f5092d03d9243c844c73332454b06c8d9ef/RecoParticleFlow/PFRecHitProducer/plugins/alpaka/PFRecHitProducerKernel.dev.cc#L62

If there is need to change the functionality to DB only, I can look into it.

Reply:
Thank you for the reminder. It's probably wise to keep the functionality to use numbers via config.